mempool/cat/reactor.go

```diff
@@ -545,31 +572,34 @@ func (memR *Reactor) processPendingSeenForSigner(signer []byte) {
 		return
 	}
 
-	expectedSeq, haveExpected := memR.sequenceExpectationForSigner(signer, true, true)
+	expectedSeq, haveExpected := memR.querySequenceFromApplication(signer)
 	if !haveExpected {
+		memR.Logger.Error("no signer found in application")
 		return
 	}
 
 	for _, entry := range entries {
-		if memR.mempool.Has(entry.txKey) {
+		if haveExpected && expectedSeq > entry.sequence {
 			memR.pendingSeen.remove(entry.txKey)
 			continue
 		}
-		if memR.requests.ForTx(entry.txKey) != 0 {
+
+		if memR.mempool.Has(entry.txKey) {
 			memR.pendingSeen.remove(entry.txKey)
 			continue
 		}
-		if expectedSeq > entry.sequence {
-			memR.pendingSeen.remove(entry.txKey)
+
+		if memR.requests.ForTx(entry.txKey) != 0 || entry.requested {
 			continue
 		}
-		if expectedSeq != entry.sequence {
+
+		// Only request if this is the next expected sequence
+		// If there's a gap, stop here and wait for the missing sequence
+		if haveExpected && entry.sequence != expectedSeq {
 			continue
 		}
 
-		if memR.tryRequestQueuedTx(entry) {
-			memR.pendingSeen.remove(entry.txKey)
-		}
+		_ = memR.tryRequestQueuedTx(entry)
```

Comment:
generally this is just a simplification where the main way that we remove pending queue things is this function instead of scattered about the rest of the mempool logic